Added tile function to ValueArray and Matrix classes.

loudinb opened the pull request:
Tile function was added to ValueArray and Matrix classes.  The tile
function generate a Matrix by tiling the source (ValueArray or Matrix)
MxN times.

Commit:
Added tile function to ValueArray and Matrix classes.

Tile function was added to ValueArray and Matrix classes.  The tile
function generate a Matrix by tiling the source (ValueArray or Matrix)
MxN times.

alejandro-isaza commented on Source/1D/ValueArray.swift:
```diff
@@ -202,6 +202,25 @@ open class ValueArray<Element: Value>: MutableLinearType, ExpressibleByArrayLite
+        return Matrix(rows: rows, columns: columns, elements: self)
+    }
+    
+    open func tile(m: Int, n: Int) -> Matrix<Element> {
```
I would remove the argument names and have instead `tile(_ m: Int, _ n Int)`

alejandro-isaza commented on Source/2D/Matrix.swift:
```diff
@@ -206,6 +206,26 @@ open class Matrix<Element: Value>: MutableQuadraticType, Equatable, CustomString
         return description
     }
+    
+    open func tile(m: Int, n: Int) -> Matrix {
```
Same here

alejandro-isaza commented on README.md:
```diff
@@ -84,6 +84,25 @@ let B = inv(A) * C // [2.0, 1.0]′
+let a = ValueArray = [1.0, 2.0]
+// Tile source array 2 times in each directon,
+// returning a 2X4 block matrix
+let A = a.tile(m: 2, n:2)
```
I think `let A = a.tile(2, 2)` is a bit more clear.

Commit:
Small syntax changed to Tile functions for clarity.

Removed argument names from Tile functions, updated appropriate test
and readme.

loudinb commented:
Thank you @aleph7 .  I have modified and committed the changes.

aleph7 commented:
Thanks!
